### ISSUE TITLE
Remove Bit.register and Bit.index, deprecated in 0.17 (released on April 01, 2021)

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit#10744]
There is code released 28 month ago that needs to be removed.
 - qiskit/circuit/bit.py:63 (`register`)
 - qiskit/circuit/bit.py:85 (`index`)
 
This includes:
- [ ] Remove deprecated code and tests (if they only check the deprecation raise)
- [ ] tag the PR as [`Changelog: Removal`](https://github.com/Qiskit/qiskit/labels/Changelog%3A%20Removal)
- [ ] create a release note in the category `upgrade:`. If you can include an example with an alternative for user to migrate to the new code, as this change might break users code.

**Comment by jakelishman** [comment]
`Bit.register` and `Bit.index` aren't really ready for removal yet - there's still code in Qiskit that uses them, and we know that there's code in backends and providers that uses them (despite them having issued warnings for years).

**Comment by 1ucian0** [comment]
Good to know! I'm labelling this `on hold`. Do you know if there are issues open in those providers?

**Comment by 1ucian0** [comment]
This is, at least, blocked on:
 * https://github.com/Qiskit/qiskit/issues/10749
 * https://github.com/Qiskit/qiskit-ibm-provider/pull/733
 * https://github.com/Qiskit/qiskit/pull/10757
 * https://github.com/qiskit-community/qiskit-aqt-provider/issues/80
 * https://github.com/Qiskit/qiskit/pull/10846

Any other issue that should be linked here? Either in Qiskit or not?

**Comment by 1ucian0** [comment]
As I continue discovering instances of `Bit.register` and `Bit.index`, I think they can stay in `~=0.45` and be removed in `~=1.0`. Setting that milestone for this one.
